external/CMakeLists.txt

```diff
@@ -7,6 +7,25 @@ if (NOT HLSL_ENABLE_DEBUG_ITERATORS)
   add_definitions(/D_ITERATOR_DEBUG_LEVEL=0)
 endif (NOT HLSL_ENABLE_DEBUG_ITERATORS)
 
+# Need DirectX-Headers module if not on windows
+if (NOT WIN32)
+  set(DXC_DIRECTX_HEADERS_DIR "${DXC_EXTERNAL_ROOT_DIR}/DirectX-Headers"
+      CACHE STRING "Location of DirectX-Headers source")
+  if (NOT DEFINED DirectX-Headers_SOURCE_DIR)
+    if (IS_DIRECTORY ${DXC_DIRECTX_HEADERS_DIR})
+      add_subdirectory(${DXC_DIRECTX_HEADERS_DIR}
```

Comment:
Currently we do not. However, I greatly prefer the approach that DirectX-Headers takes to defining dx guids as it allows the same guid structure to have the same address across multiple files which ours does not.

I chose not to take that step with this change, but I want to take it going forward when I try to better consolidate the efforts to make non-Windows platforms look a bit more like Windows for DirectX purposes, which is a goal both projects share.

Reply:
Their GUID handling is all header defined too, right? The source for it just sets a single preprocessor define then includes headers.

If we want to migrate to use it we could define that ourselves, or we depend on their static archive target. The concern I have about their static archive target is that it means filtering target dependencies through the other CMake targets. If we're only doing that on Linux it feels like we might be introducing some extra complexity. Would we shift to using the submodule headers on Windows too?

Reply:
I mean to shift to using submodule headers on Windows as it would garner us certain advantages, not the least of which being easy access to more recent headers.

You make a good point that we could just as easily recreate dxguid.cpp. Regardless of where we land on Windows using these headers, that's a sufficient argument to simplify this.